src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py

```diff
@@ -177,6 +177,10 @@ def create(dataset, session_id, target, features=None, prediction_window=100,
         else:
             dataset, validation_set = _random_split_by_session(dataset, session_id)
 
+    for feature in features:
+        if any(feat is None for feat in dataset[feature]):
+            raise _ToolkitError("Missing value (None) encountered in column " + feature + " in the training dataset. Use the SFrame's dropna function to drop rows with 'None' values in them.")
```

Comment:
Ah, this makes sense, but in my head I had naively assumed we could just check for missing values at the point when we actually process the data. This approach is much cleaner. But: can you sanity check how much overhead this applies at the beginning of training? It's nice to minimize the number of passes we take through the data.

Also, given this uniform approach, I now more agree with Krishna's point about factoring this out. You can create a function somewhere that just passes in as an argument e.g. the column name

Reply:
Factored out the code into a toolkit utility function. It is done before training. When I compare time for preprocessing time, I don't see a overhead in checking for missing values.